### ISSUE TITLE
firedux.update does not update the whole state in the reducer

### ISSUE THREAD

**Issue by uneart** [adjohnson916/firedux#19]
When calling `firedux.update( path/to/item, data)` it updates the data correctly in Firebase by **merging** it with the current data.
However, the new Redux state for `firedux.data.path.to.item` is created in the same way as for the `set()` command. That leads to a state that has **only the updated value** not the ones it was merged with.

Example:
Redux state before:
```
item: {
    title: 'Before I was named this',
    content: 'Some text as the content',
    data: '2016-01-01'
}
```

Redux state after calling `firedux.update('item', { title: 'Some new title', checked: true })`:
```
item: {
    title: 'Some new title',
    checked: true
}
```

After looking at the source code of Firedux, I suppose the reason is that the `reducer` function calls `makeFirebaseState(action, state, action.path, action.value)` for both `FIREBASE_SET` and `FIREBASE_UPDATE`.
Instead for `FIREBASE_UPDATE` it should call `makeFirebaseState(action, state, action.path, action.snapshot.val())` (the same as `FIREBASE_WATCH`). 

Please correct me, if I am wrong here :)

**Comment by adjohnson916** [comment]
@uneart You're probably right, I'll look later. Want to submit a PR with a failing unit test, and then possibly a solution as well in a subsequent commit? TDD :)

**Comment by uneart** [comment]
I fixed it the way I presume it should work. Unfortunately can't get the whole build process working on my machine right now. Sorry I'm very new to the contributing workflow on GitHub. It's embarrassing.

**Comment by uneart** [comment]
Testing a little bit, it turned out that my solution is not working, cause update() gets no snapshot value. Could the problem be the use of `updeep.constant` in `makeFirebaseState`? The [updeep docs](https://github.com/substantial/updeep#uconstantobject) show the exact same case (replacing instead of merging a value).
I tried replacing `updeep.constant(value)` with just `value` and the updating works now. But I'm not sure if it will create problems with `firedux.set()`?!

**Comment by adjohnson916** [comment]
@uneart I think I've fixed it by separating the `FIREBASE_UPDATE` vs. `FIREBASE_SET` actions and passing a flag to `makeFirebaseState` that uses just `value` in the former case but `updeep.constant(value)` in the latter when calling `updeep.updateIn`. I also adding test cases. I've released to npm as `0.6.4`. Let me know if it doesn't work or if you find any other bugs. Thanks!

**Comment by uneart** [comment]
@adjohnson916 Looks like a decent fix to me 👍 Thank you for the quick support.
